crates/core/src/sync_local.rs

```diff
@@ -61,16 +89,22 @@ pub fn sync_local(db: *mut sqlite::sqlite3, _data: &str) -> Result<i64, SQLiteEr
             "\
 -- 1. Filter oplog by the ops added but not applied yet (oplog b).
 --    SELECT DISTINCT / UNION is important for cases with many duplicate ids.
-WITH updated_rows AS (
-  SELECT DISTINCT b.row_type, b.row_id FROM ps_buckets AS buckets
-    CROSS JOIN ps_oplog AS b ON b.bucket = buckets.id
-  AND (b.op_id > buckets.last_applied_op)
-  UNION SELECT row_type, row_id FROM ps_updated_rows
-)
+WITH 
+  involved_buckets (id) AS (
+    SELECT id FROM ps_buckets WHERE ?1 IS NULL
+      OR name IN (SELECT value FROM json_each(json_extract(?1, '$.buckets')))
+  ),
+  updated_rows AS (
+    SELECT DISTINCT FALSE as local, b.row_type, b.row_id FROM ps_buckets AS buckets
+      CROSS JOIN ps_oplog AS b ON b.bucket = buckets.id AND (b.op_id > buckets.last_applied_op)
+      WHERE buckets.id IN (SELECT id FROM involved_buckets)
+    UNION SELECT TRUE, row_type, row_id FROM ps_updated_rows
+  )
 
 -- 3. Group the objects from different buckets together into a single one (ops).
 SELECT b.row_type as type,
     b.row_id as id,
+    b.local as local,
```

Comment:
Due to the GROUP BY row_type, row_id, this could select an arbitrary value if the row is present in both ps_oplog and ps_updated_rows.